### ISSUE TITLE
Running the exporter in kubernetes

### ISSUE THREAD

**Issue by runningman84** [dalance/prosafe_exporter#9]
I am trying to run this exporter in kubernetes. 

For some reason within kubernetes it does not show collect metrics from my prosafe switch. This happens  even if I run the container with hostNetwork mode.

This is the output at my workstation:
```
~ >>> curl -v "http://127.0.0.1:9493/probe?target=192.168.178.4:eth0"                                                                                                                                  
*   Trying 127.0.0.1...
* TCP_NODELAY set
* Connected to 127.0.0.1 (127.0.0.1) port 9493 (#0)
> GET /probe?target=192.168.178.4:eth0 HTTP/1.1
> Host: 127.0.0.1:9493
> User-Agent: curl/7.64.1
> Accept: */*
> 
< HTTP/1.1 200 OK
< content-length: 3323
< date: Sun, 05 May 2019 19:03:02 GMT
< 
# HELP prosafe_build_info A metric with a constant '1' value labeled by version, revision and rustversion.
# TYPE prosafe_build_info gauge
prosafe_build_info{revision="",rustversion="",version="0.2.5-pre"} 1
# HELP prosafe_error_packets_total Transfer error in packets.
# TYPE prosafe_error_packets_total gauge
prosafe_error_packets_total{port="1"} 0
prosafe_error_packets_total{port="10"} 0
prosafe_error_packets_total{port="11"} 0
prosafe_error_packets_total{port="12"} 0
..l.
```

This is the same output against my container running in the cluster:
```
$ curl -v "http://10.43.85.247:9493/probe?target=192.168.178.4:eth0"
*   Trying 10.43.85.247...
* TCP_NODELAY set
* Connected to 10.43.85.247 (10.43.85.247) port 9493 (#0)
> GET /probe?target=192.168.178.4:eth0 HTTP/1.1
> Host: 10.43.85.247:9493
> User-Agent: curl/7.52.1
> Accept: */*
> 
< HTTP/1.1 200 OK
< content-length: 293
< date: Sun, 05 May 2019 19:01:55 GMT
< 
# HELP prosafe_build_info A metric with a constant '1' value labeled by version, revision and rustversion.
# TYPE prosafe_build_info gauge
prosafe_build_info{revision="",rustversion="",version="0.2.5-pre"} 1
# HELP prosafe_up The last query is successful.
# TYPE prosafe_up gauge
prosafe_up 0
* Curl_http_done: called premature == 0
* Connection #0 to host 10.43.85.247 left intact
```

This is my helm chart for the prosafe exporter:
https://github.com/runningman84/helm-charts/tree/master/prosafe-exporter

**Comment by dalance** [comment]
prosafe_exporter must run in the same L2 network as target switches.
(This seems to be the restriction by switch's implementation. I guess this is for security.)

In your log, is your network like below?

* switch: 192.168.178.4
* your workstation eth0: 192.168.xxx.xxx
* k8s node eth0: 10.43.85.247
* k8s pod eth0: 10.43.85.247 ( the same as node because hostNetwork)

If this is correct, the cause of failure is the difference of L2 network between switch and k8s node.

**Comment by runningman84** [comment]
I run the test cluster in a virtualbox machine using a bridge network. The virtual machine is configured with the ip address 192.168.178.71. The 10.x addresses are internally in the k8s cluster used for services.

I will try to start the container on a real node in the same network. Maybe that helps...

Btw. there is another issue which makes running prosafe_exporter difficult:
https://github.com/coreos/prometheus-operator/issues/2586

It would be cool if I could configure the targets as a command line parameter and just have a static url like /metrics and each metric should contain a label for the switch name and switch port. This would allow us to monitor it using the servicemonitor....

**Comment by dalance** [comment]
I added `--target` option and released v0.2.6.
For example,

```
prosafe_exporter --target 192.168.178.4:eth0
```

and the metrics of `192.168.178.4:eth0` can be got through `/metrics`.

**Comment by runningman84** [comment]
Ok now I have also found out that you can define multiple targets using params for the service monitor:
https://github.com/runningman84/helm-charts/blob/master/prosafe-exporter/templates/servicemonitor.yaml

Funny enough I cannot get any metrics. Even if I run it without a container I get this error message in the log:
Fail to access: failed to get network interface 'eth0'

**Comment by dalance** [comment]
Does `eth0` exist? Some recent distributions have different nic name like `eno0`, `ens100`, etc.

**Comment by runningman84** [comment]
The other interface names do not work either. How can I find out the name?

**Comment by dalance** [comment]
`ifconfig` or `ip addr` can be used.

**Comment by runningman84** [comment]
Ok so the interface is from my machine and not the router itself?

**Comment by dalance** [comment]
Yes

**Comment by runningman84** [comment]
In this case it would be great if it would just probe all interfaces and save the successful interface in memory for subsequent requests. In a cluster you might have different nodes with different interface names. So it would be difficult to specify the right interface.

**Comment by dalance** [comment]
I released v0.2.7.
If `*` is used as interface name, exporter will probe all interfaces.
